client/layout/guided-tours/config-elements.js

```diff
@@ -193,12 +193,22 @@ export class Step extends Component {
 		if ( ! hasContinue && hasJustNavigated &&
 				this.isDifferentSection( lastAction.path ) ) {
 			defer( () => {
-				debug( 'Step.quitIfInvalidRoute: quitting' );
+				debug( 'Step.quitIfInvalidRoute: quitting (different section)' );
 				this.context.quit( this.context );
 			} );
-		} else {
-			debug( 'Step.quitIfInvalidRoute: not quitting' );
 		}
+
+		// quit if we have a target but cant find it
+		defer( () => {
+			const { quit } = this.context;
+			const target = targetForSlug( this.props.target );
+			if ( this.props.target && ! target ) {
```

Comment:
Interestingly in all the cases I was able to find, we duplicated any `target` in a `Continue` element in its parent `Step` component.